### ISSUE TITLE
Build System exec command - syntax key

### ISSUE THREAD

**Issue by keith-hall** [guillermooo/sublime-undocs#207]
Hi,

Sorry for all the issues, especially if you are tracking your "todo" list elsewhere and I'm potentially duplicating things, but I want to make sure everything is covered ;)

At http://docs.sublimetext.info/en/latest/reference/build_systems/exec.html, it mentions the `syntax` key, with the text `Optional. If provided, it will be used to colorize the build system’s output.`

I think it would be useful to mention that it accepts values like `scope:source.js` as well as a path to the syntax definition file `Packages/JavaScript/JSON.sublime-syntax`, and its default as per https://forum.sublimetext.com/t/new-build-system-syntax-option/14953/2

Thanks :)

**Comment by FichteFoll** [comment]
Oh, that's news to me. I didn't know `view.assign_syntax` could do that. On the other hand, the status bar display of the currently selected syntax becomes a little ... cryptic in that it shows `scope:source`.

Not sure where I would mention this though. If we had a proper and linked API documentation, I'd link to the underlying API instead for the details, but we don't have that atm. For now, I think this information is negligible.

**Comment by keith-hall** [comment]
> On the other hand, the status bar display of the currently selected syntax becomes a little ... cryptic in that it shows `scope:source`

yeah, looks like an ST3 bug to me, but it is a non-issue when dealing with output panels (i.e. build results) because only the syntax of the _view_ shows up in the status bar, not the syntax of the _panel_.

> Not sure where I would mention this though. If we had a proper and linked API documentation, I'd link to the underlying API instead for the details, but we don't have that atm.

That would indeed be awesome :)

> For now, I think this information is negligible.

fair enough :) I do think however, that finding, from the documentation, that it is possible to assign a syntax to the build results, isn't easy enough though. but I guess that is a separate issue ;)

---

but yes, after further investigation, I agree - in it's current state, I don't think it's worth documenting the "scope:" ability of `view.assign_syntax`, because it literally sets the view's `syntax` setting to that exact string, so one can't tell what syntax definition _file_ is active from `view.settings().get('syntax')` anymore.
I will log an issue at the ST3 core issues repo for it when I get chance.
